Common/Orders/Fills/EquityFillModel.cs

```diff
@@ -704,22 +713,27 @@ protected override HashSet<Type> GetSubscribedTypes(Security asset)
         }
 
         /// <summary>
-        /// Get current ask price for subscribed data
+        /// Try to get current ask price for subscribed data
         /// This method will try to get the most recent ask price data, so it will try to get tick quote first, then quote bar.
         /// If no quote, tick or bar, is available (e.g. hourly data), use trade data with preference to tick data.
+        /// If the attempt to get the current ask price does not use stale price, it's successful (returns true)  
         /// </summary>
         /// <param name="asset">Security which has subscribed data types</param>
         /// <param name="orderTime">Time the order was submitted</param>
+        /// <param name="bestEffortAskPrice">Current ask price for subscribed data</param>
         /// <param name="message">Information about the best effort, whether prices are stale or need to use trade information</param>
-        private decimal GetBestEffortAskPrice(Security asset, DateTime orderTime, out string message)
+        /// <returns>True if the best effort ask price is not stale data</returns>
+        private bool TryGetBestEffortAskPrice(Security asset, DateTime orderTime, out decimal bestEffortAskPrice, out string message)
         {
             message = string.Empty;
             BaseData baseData = null;
-            var bestEffortAskPrice = 0m;
+            bestEffortAskPrice = 0m;
 
             // Define the cut off time to get the best effort bid or ask and whether the price is stale
+            // If we don't allow fill on stale prices, the cut off time is the local order time
+            var cutOffTimeStamp = Parameters.FillOnStalePrices ? -Parameters.StalePriceTimeSpan : TimeSpan.Zero;
```

Comment:
Can discuss offline but I believe `FillOnStalePrices` should probably default to false & we should still respect `Parameters.StalePriceTimeSpan` to define when a price is considered stale or not, else the impact would be very big I think